contracts/gov/proposals/TokenSaleProposal.sol

```diff
@@ -341,13 +341,16 @@ contract TokenSaleProposal is ITokenSaleProposal, ERC1155SupplyUpgradeable {
     function _addToWhitelist(
         WhitelistingRequest calldata request
     ) internal ifTierExists(request.tierId) ifTierIsNotOff(request.tierId) {
-        _tiers[request.tierId].tierInfo.tierInfoView.uri = request.uri;
+        uint256 tierId = request.tierId;
+
+        TierInfoView storage tierInfoView = _tiers[tierId].tierInfo.tierInfoView;
+        tierInfoView.uri = request.uri;
+        tierInfoView.whitelisted = request.users.length > 0;
```

Comment:
Why save the variable to storage? Why not return in the function?

Reply:
done